pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -33,7 +50,21 @@ func getClusterName(data *clusterNameData) string {
 
 	if !data.initDone {
 		data.clusterName = config.Datadog.GetString("cluster_name")
-		// TODO: autodiscover clustername through k8s providers' API
+		// autodiscover clustername through k8s providers' API
+		if data.clusterName == "" {
+			RegisterClusterNameProviders()
+			for _, getClusterNameFunc := range ProviderCatalog {
+				clusterName, err := getClusterNameFunc()
+				if err != nil {
+					// try the next cloud provider
+					continue
+				}
+				if clusterName != "" {
+					data.clusterName = clusterName
```

Comment:
Same here, would be great to have a debug line `using cluster name %s from source %s`